lib/Unresolved.php

```diff
@@ -95,16 +94,4 @@ private function resolve(\Exception $error = null, $result = null) {
             $this->whens = $this->watchers = [];
         }
     }
-
-    private function update($progress) {
```

Comment:
This function cannot be removed -- it is used by the rebound closures here: https://github.com/amphp/amp/blob/master/lib/PrivateFuture.php#L22 ...

If the test suite still passes in its absence then I need to add more tests ;)